include/envoy/network/address.h

```diff
@@ -104,9 +105,10 @@ class Instance {
 
   /**
    * @return a human readable string for the address that represents the
-   * physical/resolved address.
+   * physical/resolved address. (This will not necessarily include port
+   * information, if applicable, since that may not be resolved until bind()).
    *
-   * This string will be in the following format:
+   * This string will be compatible with the following example formats:
```

Comment:
What do you see as the difference between the comment as stated and your change?  I see them as functionally the same and the former is a bit more clear, so I'm probably missing something :-)

Reply:
For other reviewers/The Future, from an offline conversation with Alyssa: The asString() from the new range implementation will be in the form "1.2.3.4:80-81", and I want to change the comment at this point so that it's general enough to allow that format.  Alternatively, I could edit the below examples to include range examples, but that doesn't make sense if I'm not upstreaming the range changes--what I want is a base class that allows my implementation, not one that specifies that that implementation may occur.

Reply:
I don't know if this is still the case, but it's quite possible we are trying to parse the result of `asString()` somewhere and this type of formatting would not end well. Clearly, those cases should be fixed, but just throwing it out there that you might need to do an audit.

Reply:
So I did a scan of all uses of Envoy::Network::Address::asString().  I ignored the following categories:
* Any use in tests.  I presume the tests will fail if they're doing something broken.
* Any use as a hash key.  I presume that if <address>:0 works ok for a hash key, <address>:<port1>-<port2> will also work.
* Any use in logs, or other areas that seem directly aimed at human eyes.

I came up with three locations that might be problematic, only the first of which I intend to do anything about:
* DnsResolverImpl constructor: The asString() of the resolver is passed (via csv) to ares_set_servers_ports_csv().  This is an interface problem and should be fixed (presumably by creating a string that is in the known <address>::<port>).  I will upload a diff on this CL to fix this.  Note, though, that this wouldn't cause any immediate problems, since the downstream change will only be to Address::Instance instances that refer to upstream clusters, not dns resolvers. 

* Router::RequestInfoHeaderFormatter: This appears to put the results of Instance::asString() into headers visible to the upstream instance.  However, the only Instance instances that are used for those headers are Instances referring to downstream hosts, which will never have port ranges, so this shouldn't be a problem.

*  Upstream::OriginalDstCluster::LoadBalancer::chooseHost: The concatenation of the ClusterInfo name and the downstream host asString()  is passed as an argument to the HostImpl constructor, and that value is eventually included in the headers of the request passed to upstream.  I find it hard to worry about this being parsed because the value is just jamming together a name and an asString() result, so I suspect it's a somewhat messy header whose format isn't being relied on.

Critiques of my analysis welcome.  I'll start working on the change for the DnsResolverImpl constructor.

Reply:
Uploaded new commit that handles the DnsResolverImpl case.